macros/src/lib.rs

```diff
@@ -277,3 +280,140 @@ pub fn modal(input: TokenStream) -> TokenStream {
         Err(e) => e.write_errors().into(),
     }
 }
+
+/**
+Use this macro on an impl block to implement the CommandGroup trait.
+
+It implements a `commands()` function which returns a Vec with all the commands defined by `#[poise::command]`.
+
+# Usage
+
+The following code defines a command group with two commands,
+both of which will have the `slash_command` and `prefix_command` attributes.
+
+`command_one` will have the default user_cooldown of 1000, while `command_two` overrides it to 2000.
+
+```rust
+struct MyCommands;
+
+#[poise::group(slash_command, prefix_command, user_cooldown=1000)]
+impl MyCommands {
+    /// This is a command
+    #[poise::command()]
+    async fn command_one(ctx: Context<'_>) -> Result<(), Error> {
+        // code
+    }
+
+    /// This is another command
+    #[poise::command(user_cooldown=2000)]
+    async fn command_two(ctx: Context<'_>) -> Result<(), Error> {
+        // code
+    }
+}
+```
+*/
+#[proc_macro_attribute]
+pub fn group(args: TokenStream, input_item: TokenStream) -> TokenStream {
```

Comment:
I would recommend splitting this out into a function like `group_impl` which returns a Result, so you can use `?` instead of this match spam. I implemented a similar thing in [bool_to_bitflags](https://github.com/GnomedDev/bool-to-bitflags/blob/master/src/error.rs) for inspiration.